ryot/src/bevy_ryot/sprites.rs

```diff
@@ -305,3 +325,236 @@ pub(crate) fn prepare_sprites<C: PreloadedContentAssets>(
 
     debug!("Finished preparing sprites");
 }
+
+fn load_desired_appereance_sprite<C: ContentAssets>(
+    group: AppearanceGroup,
+    id: u32,
+    frame_group_index: FixedFrameGroup,
+    direction: Option<&CardinalDirection>,
+    content_assets: &Res<C>,
+    sprites_to_be_loaded: &mut ResMut<SpritesToBeLoaded>,
+) -> Option<(
+    LoadedSprite,
+    AnimationSprite,
+    Vec<LoadedSprite>,
+    PreparedAppearance,
+)> {
+    let prepared_appearance = content_assets
+        .prepared_appearances()
+        .get_for_group(group.clone(), id)?;
+
+    let frame_group = prepared_appearance
+        .frame_groups
+        .get(frame_group_index as usize)?;
+    let sprite_info = frame_group.sprite_info.as_ref()?;
+
+    let sprites = load_sprites(
+        group.clone(),
+        &sprite_info.sprite_id,
+        content_assets,
+        sprites_to_be_loaded,
+    );
+    if sprites.len() != sprite_info.sprite_id.len() {
+        return None;
+    }
+
+    let direction_index = match direction {
+        Some(CardinalDirection::North) => 0,
+        Some(CardinalDirection::East) => 1,
+        Some(CardinalDirection::South) => 2,
+        Some(CardinalDirection::West) => 3,
+        None => 0,
+    } * sprite_info.layers() as usize;
+    let sprite = sprites.get(direction_index)?;
+
+    let animation_sprite = if let Some(animation) = sprite_info.animation.as_ref() {
+        AnimationSprite {
+            timer: Timer::from_seconds(0.1, TimerMode::Repeating),
+            sprites: sprites.clone(),
+            phase: 0,
+            initial_index: direction_index,
+            total_phases: animation.sprite_phase.len(),
+            skip: (sprite_info.layers()
+                * sprite_info.pattern_width()
+                * sprite_info.pattern_height()
+                * sprite_info.pattern_depth()) as usize,
+        }
+    } else {
+        AnimationSprite::default()
+    };
+    Some((
+        sprite.clone(),
+        animation_sprite,
+        sprites,
+        prepared_appearance.clone(),
+    ))
+}
+
+type UninitializedSpriteFilter = (
+    With<AppearanceDescriptor>,
+    Without<TextureAtlasSprite>,
+    Without<AnimationSprite>,
+);
+
+type InitializedSpriteFilter = (
+    With<AppearanceDescriptor>,
+    With<TextureAtlasSprite>,
+    With<AnimationSprite>,
+    Or<(With<LoadingAppearance>, Changed<AppearanceDescriptor>)>,
+);
+
+/// Update the sprite system, which updates the sprite appearance based on the
+/// `AppearanceDescriptor` component. It also updates the `LoadedSprites` component
+/// with the new sprites.
+/// It's meant to run every frame to update the appearance of the entities when the
+/// `AppearanceDescriptor` changes.
```

Comment:
@luan I didn't understand how will this run if the appearance descriptor change, since it doesn't use the Change<T> filter.